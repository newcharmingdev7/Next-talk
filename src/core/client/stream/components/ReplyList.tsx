import * as React from "react";
import { StatelessComponent } from "react";

import { Button } from "talk-ui/components";

import CommentContainer from "../containers/CommentContainer";
import Indent from "./Indent";

export interface ReplyListProps {
  commentID: string;
  comments: ReadonlyArray<{ id: string }>;
  onShowAll: () => void;
  hasMore: boolean;
  disableShowAll: boolean;
}

const ReplyList: StatelessComponent<ReplyListProps> = props => {
  return (
    <Indent>
      {props.comments.map(comment => (
        <CommentContainer key={comment.id} data={comment} gutterBottom />
      ))}
      {props.hasMore && (
        <Button
          id={`talk-reply-list--show-all--${props.commentID}`}
          onClick={props.onShowAll}
          disabled={props.disableShowAll}
          secondary
          invert
          fullWidth
        >
          Show All Replies
        </Button>
      )}
    </Indent>
  );
};

export default ReplyList;
