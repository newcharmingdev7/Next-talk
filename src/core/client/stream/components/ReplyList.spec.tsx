import { shallow } from "enzyme";
import { noop } from "lodash";
import React from "react";
import sinon, { SinonSpy } from "sinon";

import ReplyList, { ReplyListProps } from "./ReplyList";

it("renders correctly", () => {
  const props: ReplyListProps = {
    commentID: "comment-id",
    comments: [{ id: "comment-1" }, { id: "comment-2" }],
    onShowAll: noop,
    hasMore: false,
    disableShowAll: false,
  };
  const wrapper = shallow(<ReplyList {...props} />);
  expect(wrapper).toMatchSnapshot();
});

describe("when there is more", () => {
  const props: ReplyListProps = {
    commentID: "comment-id",
    comments: [{ id: "comment-1" }, { id: "comment-2" }],
    onShowAll: sinon.spy(),
    hasMore: true,
    disableShowAll: false,
  };

  const wrapper = shallow(<ReplyList {...props} />);
  it("renders a load more button", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("calls onLoadMore", () => {
    wrapper.find("#talk-reply-list--show-all--comment-id").simulate("click");
    expect((props.onShowAll as SinonSpy).calledOnce).toBe(true);
  });
});
