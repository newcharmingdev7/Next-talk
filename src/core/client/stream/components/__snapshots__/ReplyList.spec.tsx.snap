// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Indent>
  <Relay(CommentContainer)
    data={
      Object {
        "id": "comment-1",
      }
    }
    gutterBottom={true}
    key="comment-1"
  />
  <Relay(CommentContainer)
    data={
      Object {
        "id": "comment-2",
      }
    }
    gutterBottom={true}
    key="comment-2"
  />
</Indent>
`;

exports[`when there is more renders a load more button 1`] = `
<Indent>
  <Relay(CommentContainer)
    data={
      Object {
        "id": "comment-1",
      }
    }
    gutterBottom={true}
    key="comment-1"
  />
  <Relay(CommentContainer)
    data={
      Object {
        "id": "comment-2",
      }
    }
    gutterBottom={true}
    key="comment-2"
  />
  <withPropsOnChange(Button)
    disabled={false}
    fullWidth={true}
    id="talk-reply-list--show-all--comment-id"
    invert={true}
    onClick={[Function]}
    secondary={true}
  >
    Show All Replies
  </withPropsOnChange(Button)>
</Indent>
`;
